BrainPortal/lib/cbrain_task_generators/templates/bourreau.rb.erb

```diff
@@ -163,6 +166,14 @@ class CbrainTask::<%= name %> < <%= (descriptor['custom'] || {})['cbrain:inherit
   # in params.
   def cluster_commands #:nodoc:
 
+    # OK so this is crazy stuff, if we have a superclass
+    # that is not the default ClusterTask, we trust that it knows
+    # how to generate the commands better than we do, so we
+    # just delegate all the work to it.
+    if self.class.superclass < ClusterTask
```

Comment:
I have never seen this kind of syntax before, does this really mean "less than" the ClusterTask, does it imply that it is lower on the hierarchy of class structure?

Reply:
Yeah, the `<` and `>` comparison methods (they are methods!) are defined on class object (classes are objects!) so that you compare classes within their hierarchy. `if A < B` test that class A inherits from B (and not necessarily directly). E.g. in CBRAIN we have a class hierarchy with `Userfile < SingleFile < TextFile < LogFile`

Reply:
Oops correction

Reply:
So the above expression checks that class inherits, yet indirectly?

Reply:
Well, `self` is an instance of a particular class `Abc` for the bidsapp. So `self.class` is Abc. And `self.class.superclass` would be whatever immediate superclass is above.

The test then is `if my object's class' superclass happens to be a subclass of ClusterTask`.

Two examples: if self is an instance of SuperDuper then:

* when `ClusterTask > SuperDuper`, then if condition is false
* when `ClusterTask > Youppi > SuperDuper` then the condition is true
* when `ClusterTask > Youppi > GlopFoo > SuperDuper` then it is true too, etc.